src/sql/src/plan/statement/ddl.rs

```diff
@@ -1390,7 +1423,10 @@ fn get_encoding_inner(
                     seed
                 {
                     let mut ccsr_with_options = normalize::options(&ccsr_options);
-
+                    let url = match connector {
```

Comment:
Returning an empty string as a url to be parsed for the ccsr config doesn't seem quite right to me. Does this only want to occur if you're using an inline definition because you do this more meaningfully in `purify_csr_connector_proto`?

```diff
diff --git a/src/sql/src/plan/statement/ddl.rs b/src/sql/src/plan/statement/ddl.rs
index b62d3f53b..782a187bb 100644
--- a/src/sql/src/plan/statement/ddl.rs
+++ b/src/sql/src/plan/statement/ddl.rs
@@ -1423,16 +1423,14 @@ fn get_encoding_inner(
                     seed
                 {
                     let mut ccsr_with_options = normalize::options(&ccsr_options);
-                    let url = match connector {
-                        mz_sql_parser::ast::CsrConnector::Inline { url: uri } => uri,
-                        mz_sql_parser::ast::CsrConnector::Reference { .. } => "",
-                    };
-                    // We validate here instead of in purification, to match the behavior of avro
-                    let _ccsr_config = kafka_util::generate_ccsr_client_config(
-                        url.parse()?,
-                        &kafka_util::extract_config(&mut normalize::options(with_options))?,
-                        &mut ccsr_with_options,
-                    )?;
+                    if let mz_sql_parser::ast::CsrConnector::Inline { url } = connector {
+                        // We validate here instead of in purification, to match the behavior of avro
+                        let _ccsr_config = kafka_util::generate_ccsr_client_config(
+                            url.parse()?,
+                            &kafka_util::extract_config(&mut normalize::options(with_options))?,
+                            &mut ccsr_with_options,
+                        )?;
+                    }
```

Reply:
Oh dammit, this should not have slipped through. I am actually completely stunned that this even passes tests, I had written it this way originally to make the `match` arms type check while I was fleshing out other details and all of the other places I took a similar shortcut and forgot to fix emerged in testing quite rapidly. 

I will fix this but I also need to figure out why it did not get caught.